cpp/ql/src/semmle/code/cpp/rangeanalysis/SimpleRangeAnalysis.qll

```diff
@@ -249,6 +255,14 @@ private predicate defDependsOnDef(
     exprDependsOnDef(assignSub.getRValue(), srcDef, srcVar)
   )
   or
+  exists(UnsignedAssignMulExpr assignMul, RangeSsaDefinition nextDef |
+    def = assignMul and
+    assignMul.getLValue() = nextDef.getAUse(v)
+  |
+    defDependsOnDef(nextDef, v, srcDef, srcVar) or
```

Comment:
I'm confused about why we need this recursive call (both here and for the other `AssignExpr` subclasses).

Reply:
Well spotted. I don't think the desugared version (`x = x * e`) does this recursion. I've pushed a commit to remove the recursion (but still take dependency information from the LHS).

Reply:
I've started https://jenkins.internal.semmle.com/job/Changes/job/CPP-Differences/1321/ for just this change (edc5e5f).